### DISCUSSION TITLE
Build Failed

### DISCUSSION THREAD

**Discussion by soeryobadja** [tenacityteam/tenacity-legacy#484] [Q&A] [answered]
Hello,
My background was not in computer programming, but I need to use Audacity without so-called spyware related to their new T&C which brings me to this project. 
I was following the build guide for macOS, but failed in the last step of building the Audacity app. Here is what the information.

`
Command line invocation:
    /Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild -project Audacity.xcodeproj build -target ALL_BUILD -parallelizeTargets -configuration Release -hideShellScriptEnvironment

User defaults from command line:
    HideShellScriptEnvironment = YES
    IDEPackageSupportUseBuiltinSCM = YES

note: Using new build system
note: Building targets in parallel
note: Planning build
note: Analyzing workspace
note: Constructing build description
note: Build preparation complete
error: Multiple commands produce '/Users/mynuc/audacity/bld/bin/Release/Audacity.app/Contents/MacOS':
1) CreateBuildDirectory /Users/mynuc/audacity/bld/bin/Release/Audacity.app/Contents/MacOS
2) Target 'Audacity' (project 'Audacity') has create directory command with output '/Users/mynuc/audacity/bld/bin/Release/Audacity.app/Contents/MacOS'

CreateBuildDirectory /Users/mynuc/audacity/bld
    cd /Users/mynuc/audacity/bld/Audacity.xcodeproj
    builtin-create-build-directory /Users/mynuc/audacity/bld

warning: duplicate output file '/Users/mynuc/audacity/bld/bin/Release/Audacity.app/Contents/MacOS' on task: MkDir /Users/mynuc/audacity/bld/bin/Release/Audacity.app/Contents/MacOS (in target 'Audacity' from project 'Audacity')
** BUILD FAILED **
`

Please help.

Thanks.

**Comment by poyribn** [comment]
Unfortunately I can't help you installing Tenacity,
but in þe meantime, you can run audacity in a Sandbox.

It will be different in MacOS, but in Linux I do þis to run audacity without Muse Group spying:
I use þe command ``firejail net=none audacity`` to run audacity without access to the internet. For safety, I set þis as the alias overriding the regular ``audacity`` command.

If you're interested, you can watch þis tenacity repository to be notified about tenacity updates. Once þe first release version gets released, it will be much easier to install tenacity on MacOS and all other operating systems.

**Comment by crsib** [comment]
You have likely missed `-T buildsystem=1` while configuring the project. This issue is known and it will be addressed in Audacity in the future, I'm not sure if Tenacity maintainers will choose to implement the fix as well.

Please note, that Audacity 3.0.2 has no networking at all, and update checks can be easily disabled before the first check. Also, if you choose to use your own Audacity build - there will be no networking as well.

**Reply by Be-ing** [reply]
We have a major overhaul of the build system which is almost done in https://github.com/tenacityteam/tenacity/pull/228. The un/underdocumented CMake options like `buildsystem` have been removed and are no longer needed.

**Reply by crsib** [reply]
1. I don't see any changes to the `Wrapper` target, and they are required to support the "New" build system. In short, there should be a target that creates the `MacOS` directory, and `Wrapper`/`Tenacity` should depend on it. Probably I've missed it, in this case, I'm sorry.
2. `-T buildsystem=1` is well documented: https://cmake.org/cmake/help/v3.19/generator/Xcode.html#xcode-build-system-selection.
3. You won't notice this problem with CI builds (for now) because of the default XCode version used by GitHub actions.

PS Your PR doesn't build out-of-the-box on macOS for me because I don't have Brew and have no intentions to install Brew. It's the easiest way to wreck the system, and it doesn't play well on M1 Macs. I don't think that you really care or should care about it, though.

**Reply by unbeatable-101** [reply]
> it doesn't play well on M1 Macs

I haven't noticed any issues on my M1 compared to my old Intel Mac

**Reply by Be-ing** [reply]
The XCode CMake generator is incompatible with compiler caching so we already stopped using it, so I don't think the `buildsystem` argument is relevant anymore.

**Reply by Be-ing** [reply]
> PS Your PR doesn't build out-of-the-box on macOS for me because I don't have Brew and have no intentions to install Brew. It's the easiest way to wreck the system, and it doesn't play well on M1 Macs. I don't think that you really care or should care about it, though.

Homebrew is only a temporary hack until the wxWidgets package in vcpkg is fixed. macOS builds with wxWidgets built manually though if someone really doesn't want to use Homebrew.

**Reply by Be-ing** [reply]
CI builds in #228 are no longer using wxWidgets from Homebrew because Homebrew builds for a minimum macOS version of 10.15.

**Comment by Be-ing** [comment, marked as answer]
#228 has been merged. Please try again and refer to the rewritten [BUILDING.md](https://github.com/tenacityteam/tenacity/blob/master/BUILDING.md) for build instructions. The XCode CMake generator is untested and slow. Using Ninja and installing ccache are recommended.
